modules/sdk-coin-hbar/src/hbar.ts

```diff
@@ -592,4 +604,78 @@ export class Hbar extends BaseCoin {
 
     return { txId: transactionResponse.transactionId.toString(), status: transactionReceipt.status.toString() };
   }
+
+  /** @inheritDoc */
+  async deriveKeyWithSeed({ key, seed }: { key: string; seed: string }): Promise<{ key: any; derivationPath: string }> {
+    const hdTree = await Ed25519Bip32HdTree.initialize();
+
+    const derivationPath = getDerivationPath(seed);
+
+    if (key.startsWith('rpub')) {
+      // generic code that could be moved out of this function
+      const [pub, cc] = key.replace('rpub', '').split(':');
+
+      const pk = bigIntFromBufferLE(Buffer.from(pub, 'hex'));
+      const chaincode = bigIntFromBufferBE(Buffer.from(cc, 'hex'));
+
+      const pubKeychain = { pk, chaincode };
+      const derivedKeychain = hdTree.publicDerive(pubKeychain, derivationPath);
+      const derivedPub = bigIntToBufferLE(derivedKeychain.pk).toString('hex');
+
+      // coinspecific code
+      const derivedHbarPub = new HbarKeyPair({ pub: derivedPub }).getKeys().pub;
+
+      return { key: derivedHbarPub, derivationPath };
+    } else if (key.startsWith('rprv')) {
+      await Eddsa.initialize();
+      // generic code that could be moved out of this function
+      const [prv, cc, prfx] = key.replace('rprv', '').split(':');
+      const chaincode = bigIntFromBufferBE(Buffer.from(cc, 'hex'));
+      const sk = bigIntFromBufferLE(Buffer.from(prv, 'hex'));
+      const prefix = bigIntFromBufferBE(Buffer.from(prfx, 'hex'));
+
+      // coinspecific code to get the public key
+      const rootKeyPair = new HbarKeyPair({ prv: prv });
+      const pub = rootKeyPair.getKeys(true).pub;
+
+      // generic code that could be moved out of this function
+      const pk = bigIntFromBufferLE(Buffer.from(pub, 'hex'));
+      const derivedKeychain = hdTree.privateDerive({ sk, pk, chaincode, prefix }, derivationPath);
+      const derivedPrv = bigIntToBufferLE(derivedKeychain.sk).toString('hex');
+      // const derivedPub = bigIntToBufferLE(derivedKeychain.pk).toString('hex');
+
+      // coinspecific code
+      const derivedHbarPrv = new HbarKeyPair({ prv: derivedPrv }).getKeys().prv;
```

Comment:
i thought we figured out we need to append a pub to the derivedPrv for Hbar to use this prv as we expect

Reply:
This derivedPrv is the "hbar" formatted prv, appending the pub to the doesnt work
Also, appending the pub means that we always used the prv as a seed based on what we saw where the checks expect prv length === 64 and the seed === 32

Reply:
so derivedPrv above seems to me like it's not formatted for any coin yet until line 648. I remember when we looked at the code we found out that when u pass a 32 byte prv to HbarKeyPair it uses as a seed. That's not what we want. We want it to use it as is. So for that u had to append the pub key cause when prv is 64 bytes the first 32 bytes are used as the prv and the last 32 bytes are used as the pub key

Reply:
Sorry, editted the msg, at this point we only care about the prv and the return of this is the formatted prv, is the same result if you use prv or prv+pub (my guess cause of the KeyPair implementation)